autopilot/migrator/accounts.go

```diff
@@ -0,0 +1,94 @@
+package migrator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	rhpv4 "go.sia.tech/core/rhp/v4"
+	"go.sia.tech/core/types"
+	"go.sia.tech/renterd/api"
+	"go.sia.tech/renterd/internal/gouging"
+	"go.sia.tech/renterd/internal/locking"
+	rhp3 "go.sia.tech/renterd/internal/rhp/v3"
+)
+
+const (
+	defaultRevisionFetchTimeout = 30 * time.Second
+
+	lockingPrioritySyncing = 30
+)
+
+func (m *migrator) FundAccount(ctx context.Context, fcid types.FileContractID, hk types.PublicKey, desired types.Currency) error {
```

Comment:
Maybe we can move these methods into the `accounts` to avoid duplicating this logic here and in the worker.

Reply:
Both this and the migrator dependencies not being properly injected boil down to a nasty catch 22 we have between our host and accounts manager. There's a couple of solutions but both should probably be tackled in a follow up, it's also not really v2 critical. They are essentially dependant on each other, which makes creating them a bit of a pain obviously. 

Out of all the options there's two I like best, a) we refactor the account manager to have an account store, both host and account manager can get the store and the host manager no longer needs the account manager or b) we merge the account manager and the host manager into one because everyone likely needs both anyway, e.g. the host manager also performs account refills...